samples/net/mqtt_sn_publisher/Kconfig

```diff
@@ -5,12 +5,21 @@
 
 mainmenu "MQTT-SN sample application"
 
+config NET_SAMPLE_MQTT_SN_STATIC_GATEWAY
+	bool "Whether to statically define the Gateway. Will use Search procedure if False."
+
 config NET_SAMPLE_MQTT_SN_GATEWAY_IP
-	string "IP of the MQTT-SN gateway"
+	string "IP of the MQTT-SN gateway. Only used if NET_SAMPLE_MQTT_SN_STATIC_GATEWAY=n."
```

Comment:
You could put `depends on` here for the `NET_SAMPLE_MQTT_SN_STATIC_GATEWAY` to hide the symbol so user would not be able to set the value.

Reply:
So I did try that. In samples/net/mqtt_sn_publisher/src/udp.c:141 I use `if (IS_ENABLED(CONFIG_NET_SAMPLE_MQTT_SN_STATIC_GATEWAY)) {` with `CONFIG_NET_SAMPLE_MQTT_SN_GATEWAY_IP` referenced as requested earlier. If I make `NET_SAMPLE_MQTT_SN_GATEWAY_IP` depend on `NET_SAMPLE_MQTT_SN_STATIC_GATEWAY` then when `NET_SAMPLE_MQTT_SN_STATIC_GATEWAY=n`, `NET_SAMPLE_MQTT_SN_GATEWAY_IP` is undefined and build fails. Should I put a `#ifdef` around that part as well or is there a `IS_ENABLED` analog for if a CONFIG is defined at all?

Reply:
You could do for example like this
```
#ifdef CONFIG_NET_SAMPLE_MQTT_SN_STATIC_GATEWAY
#define SAMPLE_GW_IP CONFIG_NET_SAMPLE_MQTT_SN_STATIC_GATEWAY_IP
#else
#define SAMPLE_GW_IP ""
#endif 
```
and then use `SAMPLE_GW_IP` when referring to the config value.
Alternatively there is also `IF_EXISTS()` macro that could be used here.